src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.FruitOperationDao;
+import core.basesyntax.model.FruitOperation;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private FruitOperationDao fruitOperationDao;
+    private OperationStrategy strategy;
+
+    public ShopServiceImpl(FruitOperationDao fruitOperationDao, OperationStrategy strategy) {
+        this.fruitOperationDao = fruitOperationDao;
+        this.strategy = strategy;
+    }
+
+    @Override
+    public void changeQuantityStore(List<FruitOperation> fruits) {
+        for (FruitOperation fruit : fruits) {
+            if (fruit.getOperation() == FruitOperation.Operation.BALANCE) {
+                fruitOperationDao.add(fruit);
+            }
+            FruitOperation currentFruit = fruitOperationDao.get(fruit.getFruit());
+            int newValueFrom = strategy.get(fruit.getOperation())
```

Comment:
The `strategy.get(fruit.getOperation())` call can potentially return `null` if the operation is not supported. You should handle this case to avoid a `NullPointerException`. Consider throwing an informative exception if the operation is not found in the strategy map.